insonmnia/dwh/server_test.go

```diff
@@ -18,47 +26,193 @@ import (
+			fmt.Println(err)
+		}
+		if err := cli.ContainerRemove(ctx, containerID, types.ContainerRemoveOptions{}); err != nil {
+			fmt.Println(err)
+		}
+		if err := cli.Close(); err != nil {
+			fmt.Println(err)
+		}
+		os.Exit(testsReturnCode)
+	}()
+
+	if err := setupDB(); err != nil {
+		fmt.Println(err)
+		return
 	}
 
-	monitorDWH, err = getTestDWH(testMonitorDBPath)
+	globalDWH, err = getTestDWH(getConnString(globalDBName, dbUser, dbUserPassword))
 	if err != nil {
 		fmt.Println(err)
-		os.Remove(testMonitorDBPath)
-		os.Exit(1)
+		return
+	}
+
+	monitorDWH, err = getTestDWH(getConnString(monitorDBName, dbUser, dbUserPassword))
+	if err != nil {
+		fmt.Println(err)
+		return
 	}
 
-	retCode := m.Run()
-	globalDWH.db.Close()
-	os.Remove(testDBPath)
-	os.Remove(testMonitorDBPath)
-	os.Exit(retCode)
+	testsReturnCode = m.Run()
 }
 
-func getTestDWH(dbPath string) (*DWH, error) {
+func startPostgresContainer(ctx context.Context) (cli *client.Client, containerID string, err error) {
+	cli, err = client.NewEnvClient()
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to setup Docker client: %s", err)
+	}
+
+	reader, err := cli.ImagePull(ctx, "docker.io/library/postgres", types.ImagePullOptions{})
+	if err != nil {
+		cli.Close()
+		return nil, "", fmt.Errorf("failed to pull postgres image: %s", err)
+	}
+	io.Copy(os.Stdout, reader)
+
+	containerCfg := &container.Config{
+		Image:        "postgres",
+		ExposedPorts: nat.PortSet{"5432": struct{}{}},
+	}
+	hostCfg := &container.HostConfig{
+		PortBindings: map[nat.Port][]nat.PortBinding{
+			nat.Port("5432"): {{HostIP: "localhost", HostPort: postgresPort}},
+		},
+	}
+	resp, err := cli.ContainerCreate(ctx, containerCfg, hostCfg, nil, "dwh-postgres-test")
+	if err != nil {
+		cli.Close()
+		return nil, "", fmt.Errorf("failed to create container: %s", err)
+	}
+
+	if err := cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+		cli.Close()
+		return nil, "", fmt.Errorf("failed to start container: %s", err)
+	}
+
+	return cli, resp.ID, nil
+}
+
+func checkPostgresReadiness(containerID string) error {
+	var (
+		err        error
+		numRetries = 10
+	)
+	for ; numRetries > 0; numRetries-- {
+		cmd := exec.Command("docker", "exec", containerID, "pg_isready")
+		out, err := cmd.CombinedOutput()
+		if err == nil {
+			if strings.Contains(string(out), "accepting connections") {
+				return nil
+			}
+		}
+		fmt.Printf("postgres container not ready, %d retries left\n", numRetries)
+		time.Sleep(time.Second)
+	}
+
+	return fmt.Errorf("failed to connect to postgres container: %v", err)
+}
+
+func setupDB() error {
+	db, err := sql.Open("postgres", serviceConnString)
+	if err != nil {
+		return fmt.Errorf("failed to connect to template1: %s", err)
+	}
+	defer db.Close()
+
+	for _, dbName := range []string{globalDBName, monitorDBName} {
+		if _, err := db.Exec(fmt.Sprintf("DROP DATABASE IF EXISTS %s", dbName)); err != nil {
```

Comment:
For those who'll review later: prepared statements will be more clear here, but the postgresql [does not support](https://github.com/lib/pq/issues/694) it for services statements.